cmd/storagenode-updater/main.go

```diff
@@ -217,9 +232,41 @@ func update(ctx context.Context, nodeID storj.NodeID) (err error) {
 	return nil
 }
 
+func renameStoragenode(currentVersion version.SemVer) error {
+	extension := filepath.Ext(runCfg.BinaryLocation)
+	dir := filepath.Dir(runCfg.BinaryLocation)
+	backupExec := filepath.Join(dir, runCfg.ServiceName+".old."+currentVersion.String()+extension)
+
+	if err := os.Rename(runCfg.BinaryLocation, backupExec); err != nil {
+		return errs.Wrap(err)
+	}
+	return nil
+}
+
+func renameUpdater(_ version.SemVer) error {
+	updaterBinName := os.Args[0]
+	extension := filepath.Ext(updaterBinName)
+	dir := filepath.Dir(updaterBinName)
+	base := filepath.Base(updaterBinName)
+	base = base[:len(base)-len(extension)]
+	backupExec := filepath.Join(dir, base+".old"+extension)
+
+	if err := os.Rename(updaterBinName, backupExec); err != nil {
+		return errs.Wrap(err)
+	}
+	return nil
+}
+
+func parseDownloadURL(template string) string {
+	url := strings.Replace(template, "{os}", runtime.GOOS, 1)
+	url = strings.Replace(url, "{arch}", runtime.GOARCH, 1)
+	return url
+}
+
 func binaryVersion(location string) (version.SemVer, error) {
-	out, err := exec.Command(location, "version").Output()
+	out, err := exec.Command(location, "version").CombinedOutput()
 	if err != nil {
+		log.Printf("out %s", string(out))
```

Comment:
Do we actually need to print `out`?

If yes and you use the `%s` verb then there isn't a need for casting the byte slice to string.

Reply:
I left print for debugging purposes but I removed casting.

Reply:
printing was fine, I was referencing the `"out"` word in the print.

Reply:
I changed it to `command output: %s`